server.go

```diff
@@ -462,11 +462,85 @@ func (p *sshFxpOpenPacket) respond(svr *Server) responsePacket {
 		osFlags |= os.O_EXCL
 	}
 
-	f, err := os.OpenFile(svr.toLocalPath(p.Path), osFlags, 0o644)
+	name := svr.toLocalPath(p.Path)
+	mode := os.FileMode(0o644)
+
+	var applyAttrs []func(*os.File) error
+
+	// Both `sshFileXferAttrPermissions` and `sshFileXferAttrACmodTime` are set
+	// by e.g. `sftp`. Just in case, we handle all other cases as well.
+	// Only apply for newly created files.
+	var err error
+	useAttrs := true
+	if _, err := os.Stat(name); err == nil {
+		useAttrs = false
+	}
+	if b, ok := p.Attrs.([]byte); useAttrs && ok {
+		if (p.Flags & sshFileXferAttrSize) != 0 {
+			var size uint64
+			if size, b, err = unmarshalUint64Safe(b); err == nil {
+				applyAttrs = append(applyAttrs, func(f *os.File) error {
+					return f.Truncate(int64(size))
```

Comment:
I guess there is technically a possibility of overflow here. Converting from a `uint64` to an `int64`.

Reply:
This follows the existing implementation elsewhere. I think this is mostly theoretical though, either buggy clients or files exceeding 9000 PB?

Reply:
Yeah, probably safe, just noticed it. (inb4 new 8 EiB of filespace ought to be enough for anybody.)

Reply:
This was removed anyway (only the old existing code for stat packets is left).